### ISSUE TITLE
Horrible memory leak

### ISSUE THREAD

**Issue by netikras** [Keruspe/GPaste#227]
Hello,

I've installed gpaste yesterday from LinuxMint (ubuntu) repos. I'm on LM Sarah. Today I've noticed mouse cursor is getting slower and eventually the whole UI froze meaning there's probably no more memory for the UI. After OOMK nuked chrome and gitkraken I managed to get into TTY1 and run ps aux | sort -nrk4 | head. gpaste-daemon is at the very top with 8.6GB RSS usage. I'm currently on TTY1 hence I'm not copying anything. Any time OOMK nukes some process gpaste-daemon consumes all available memory there is. Since last start of the gpaste I've copyed up to 20 pictures and up to 50 text strings.

I have not rebooted the PC yet but I cannot prevent OOMKiller from killing the daemon. Also w/o memory TTY1 is terribly slow. So if there is anything I can do ATM to collect info while daemon is still running tell me now and I will try to. I can still kill my IDE to free up some memory and run maybe one or two commands until it's consumed by gpaste.

I cannot tell the exact version of gpaste yet - gpaste-client --version is still hanging for 10 minutes now.

UPDATE:
installing gpaste on nearly identical build gives me GPaste 3.18.6 version. So I'm assuming my workstation has the same one.

UPDATE2:
gpaste-daemon is now at 10G VSZ and 9.3G RSS. CPU%: 0.3.

**Comment by netikras** [comment]
Workstation eventually froze completely... I've gotta force-reboot it now. Damn it hurts :( 

Anything I can gather post-mortem?

**Comment by Keruspe** [comment]
Which size have you configured for your history?
Do you copy lots of images?

**Comment by netikras** [comment]
![image](https://user-images.githubusercontent.com/2045031/35437564-9bcadd02-029b-11e8-995c-0b6e95a35e6a.png)

![image](https://user-images.githubusercontent.com/2045031/35437582-b10cd436-029b-11e8-9de9-9294309e3016.png)


here you go. There's no way it's 10GB+. As stated before, since yesterday I have copied up to 20 images - no more.

**Comment by Keruspe** [comment]
This is a really old version of GPaste. If you disable images support, can
you reproduce?

**Comment by netikras** [comment]
well that's what I got from repos :) 

Dunno if it can be reproduced w/o images support. I logged an issue as soon as I saw the leak. 

Shall I just forget about it and install the newest version there is (not from repos)? Or do you prefer me to try and see if the leak can be reproduced w/o images in the one I already have?

**Comment by Keruspe** [comment]
If you can reproduce the leak without images, there definitely is something
fishy going on. If you cannot, there was an history of leaks in the images
support which is supposedly fixed in newer versions.

Btw, when this happens, « gpaste-client daemon-reexec » might help

**Comment by netikras** [comment]
Allright then, I'll leave current version with images support disabled until Monday morning (UTC). If anything pops out I'll let you know. 

Thanks

**Comment by netikras** [comment]
MEM usage looks fine.

**Comment by netikras** [comment]
Also I have a hunch on why only a 2 year-old version of gpaste is available in PPAs... I bet it has smth do do with those:

```
Requested 'glib-2.0 >= 2.52.0' but version of GLib is 2.48.2
Requested 'gobject-2.0 >= 2.52.0' but version of GObject is 2.48.2
Requested 'gio-2.0 >= 2.52.0' but version of GIO is 2.48.2
```

:) 2.48 comes with OS which has been released... yep, that's right - 2 yrs ago :)  https://github.com/GNOME/glib/releases/tag/2.48.2

**Comment by Keruspe** [comment]
FYI, #230 allowed me to finally get where the leak came from, and I'm currently in the process of backporting the fix to all versions up to 3.18 :)

**Comment by maltewhiite** [comment]
I am experiencing that the app starts to lag more and more, the longer I have it open. After 6 hours, it takes 5 seconds to input a single character into commit message field.
